tests/tests/test-transaction-prechecks.ts

```diff
@@ -0,0 +1,75 @@
+import { expect } from "chai";
+
+import { createAndFinalizeBlock, customRequest, describeWithMoonbeam } from "./util";
+import {
+  FIRST_CONTRACT_ADDRESS,
+  GENESIS_ACCOUNT,
+  GENESIS_ACCOUNT_PRIVATE_KEY,
+  TEST_CONTRACT_ABI,
+  TEST_CONTRACT_BYTECODE,
+} from "./constants";
+
+let tx;
+
+describeWithMoonbeam(
+  "Moonbeam RPC (Transaction validity pre-checks)",
+  `simple-specs.json`,
+  (context) => {
+    it("should not return error on the same block", async function () {
+      tx = await context.web3.eth.accounts.signTransaction(
+        {
+          from: GENESIS_ACCOUNT,
+          data: TEST_CONTRACT_BYTECODE,
+          value: "0x00",
+          gasPrice: "0x01",
+          gas: "0x100000",
+        },
+        GENESIS_ACCOUNT_PRIVATE_KEY
+      );
+      // `a` is a new submitted transaction.
+      let a = await customRequest(context.web3, "eth_sendRawTransaction", [tx.rawTransaction]);
+      // `b` is already known at Rpc-level, and is not submitted to the pool.
+      let b = await customRequest(context.web3, "eth_sendRawTransaction", [tx.rawTransaction]);
+
+      expect(a.result).to.be.equal(
+        "0xe87ed993e4d186748404a52a2d13612eef8356331f30fa6b3fb9bc2c16be2e9c"
+      );
+      // Before adding the checks, bellow assertion would fail with `submit transaction to
+      // pool failed: Pool(AlreadyImported(Any))`.
+      expect(b.result).to.be.equal(
+        "0xe87ed993e4d186748404a52a2d13612eef8356331f30fa6b3fb9bc2c16be2e9c"
+      );
```

Comment:
So the desired behaviour on Ethereum is that users can submit the same transaction over and over an the node just acts like it is the first time? I guess that makes some sense if the tx is still in the pool.

But then, why are we treating this as a hotfix instead of making a proper PR to frontier?